SpatialDE/_internal/score_test.py

```diff
@@ -273,3 +273,102 @@ def _grad_negative_negbinom_loglik(params, counts, sizefactors):
             + (counts - mus) / one_alpha_mu
         )  # d/d_alpha
         return -tf.convert_to_tensor((grad0, grad1))
+
+
+class NormalScoreTest(ScoreTest):
+    @dataclass
+    class NullModel(ScoreTest.NullModel):
+        mu: tf.Tensor
+        sigma: tf.Tensor
+
+    def __init__(
+        self,
+        sizefactors: tf.Tensor,
+        omnibus: bool = False,
+        kernel: Optional[Union[Kernel, List[Kernel]]] = None,
+    ):
+        self.sizefactors = tf.squeeze(tf.cast(sizefactors, tf.float64))
+        if tf.rank(self.sizefactors) > 1:
+            raise ValueError("Size factors vector must have rank 1")
+
+        yidx = tf.cast(tf.squeeze(tf.where(self.sizefactors > 0)), tf.int32)
+        if tf.shape(yidx)[0] != tf.shape(self.sizefactors)[0]:
+            self.sizefactors = tf.gather(self.sizefactors, yidx)
+        else:
+            yidx = None
+        super().__init__(omnibus, kernel, yidx)
+
+    def _fit_null(self, y: tf.Tensor) -> NullModel:
+        scaledy = tf.cast(y, tf.float64) / self.sizefactors
+        res = minimize(
+            lambda *args: self._negative_normal_loglik(*args).numpy(),
+            x0=[
+                tf.reduce_mean(scaledy),
+                tf.zeros_like(y),
+            ],
+            args=(tf.cast(y, tf.float64), self.sizefactors),
+            jac=lambda *args: self._grad_negative_normal_loglik(*args).numpy(),
+            method="bfgs",
+        )
+        mu = res.x[0] * self.sizefactors
+        sigma = tf.exp(res.x[1])
+        return self.NullModel(mu, sigma)
+
+    def _test(
+        self, y: tf.Tensor, nullmodel: NullModel
+    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor, tf.Tensor]:
+        return self._do_test(
+            self._K,
+            to_default_float(y),
+            to_default_float(nullmodel.sigma),
+            to_default_float(nullmodel.mu),
+        )
+
+    @staticmethod
+    @tf.function(experimental_compile=True)
+    def _do_test(
+        K: tf.Tensor, rawy: tf.Tensor, sigma: tf.Tensor, mu: tf.Tensor
+    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor, tf.Tensor]:
+        W = 1 / sigma**2 # Delta W^-1
+        stat = 0.5 * tf.reduce_sum(
+            (rawy - mu) * W * tf.tensordot(K, W * (rawy - mu), axes=(-1, -1)), axis=-1
+        )
+
+        P = tf.linalg.diag(W) - W[:, tf.newaxis] * W[tf.newaxis, :] / tf.reduce_sum(W)
+        PK = W[:, tf.newaxis] * K - W[:, tf.newaxis] * ((W[tf.newaxis, :] @ K) / tf.reduce_sum(W))
+        trace_PK = tf.linalg.trace(PK)
+        e_tilde = 0.5 * trace_PK
+        I_tau_tau = 0.5 * tf.reduce_sum(PK * PK, axis=(-2, -1))
+        I_tau_theta = 0.5 * tf.reduce_sum(PK * P, axis=(-2, -1))
+        I_theta_theta = 0.5 * tf.reduce_sum(tf.square(P), axis=(-2, -1))
+        I_tau_tau_tilde = I_tau_tau - tf.square(I_tau_theta) / I_theta_theta
+
+        return stat, e_tilde, I_tau_tau_tilde
+
+    @staticmethod
+    @tf.function(experimental_compile=True)
+    def _negative_normal_loglik(params, observations, sizefactors=None):
```

Comment:
why is this called negative_normal? That looks like just a Normal distribution to me.

Reply:
It's negative log likelihood for the Normal distribution!

Reply:
That's true! I've made the changes